src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs

```diff
@@ -652,6 +759,10 @@ internal void Save(string filePath, bool isBackup = false, EngineController engi
                 {
                     saveContent = jo.ToString();
                 }
+
+                JsonRepresentation = JObject.Parse(saveContent);
+                GraphId = JsonRepresentation.Properties().First(p => p.Name == "Uuid").Value.ToString();
```

Comment:
1. Why are you introducing a new property GraphId on the workspace view model- you already have access to the Uuid of the graph on the model - since you only call `HasSubstantialChecksum` after you save is called. Does the Model's UUID and the serialized UUID get out of sync? If so that seems like a bug that is better to fix than a introduce yet another ID property which itself get out of sync and be confusing to other developers. I am already confused.

Reply:
> Why are you introducing a new property GraphId on the workspace view model- you already have access to the Uuid of the graph on the model - since you only call `HasSubstantialChecksum` after save is called.
> 
> Does the Model's UUID and the serialized UUID get out of sync? If so that seems like a bug that is better to fix than a introduce yet another ID property which itself get out of sync and be confusing to other developers. I am already confused.

Yup, those values were diferent when we save a new graph due to all the guids are regenerated on the fly, probably at that point it wasn't so important because there are not more interactions afer the saving, but now we have it, fixed now.